src/basic/stat-util.c

```diff
@@ -275,18 +275,21 @@ int stat_verify_regular(const struct stat *st) {
         if (!S_ISREG(st->st_mode))
                 return -EBADFD;
 
+        if (check_removed && st->st_nlink <= 0)
+                return -EIDRM;
```

Comment:
hmm, instead of adding a "full" function, we could just return 0 in case of a deleted file and 1 in case of an existing file. Then, callers could handle this on their own?

(just and idea, up to you, "full" is fine too)

Reply:
(but as dicussed below, maybe don#t add this to stat_verify_regular() at all, but instead add fd_verify_linked() instead, see discussion)